templates/support/SecurityClient.swift.hbs

```diff
@@ -0,0 +1,361 @@
+//  
+//  {{>frag/generatedBy}}
+//
+
+import Foundation
+
+public protocol SecurityClient {
+    /** An API request received a 401 response */
+    func authenticationFailed(securityScheme: SecurityScheme, scopes: [String]?) async throws
+    
+    /** An authorization header to the request */
+    func authorize(securityScheme: SecurityScheme, scopes: [String]?, request: URLRequest) async throws -> URLRequest
+}
+
+/** Delegates security requiements to the appropriate SecurityClient. */
+public class SecurityClientController: SecurityClient {
+    public func authenticationFailed(securityScheme: SecurityScheme, scopes: [String]?) async throws {
+        if let client = clients[securityScheme] {
+            try await client.authenticationFailed(securityScheme: securityScheme, scopes: scopes)
+        } else {
+            throw APIError.securitySchemeNotConfigured(securityScheme)
+        }
+    }
+
+    public func authorize(securityScheme: SecurityScheme, scopes: [String]?, request: URLRequest) async throws -> URLRequest {
+        if let client = clients[securityScheme] {
+            return try await client.authorize(securityScheme: securityScheme, scopes: scopes, request: request)
+        } else {
+            return request
+        }
+    }
+    
+    var clients: [SecurityScheme: SecurityClient] = [:]
+    
+    public func setClient(securityScheme: SecurityScheme, client: SecurityClient) {
+        self.clients[securityScheme] = client
+    }
+
+    public init(clients: [SecurityScheme : SecurityClient] = [:]) {
```

Comment:
Is this normal to put the constructor at the bottom of the class? I thought they normally went near the top!